src/dataExplorer/components/DBRPSelector.tsx

```diff
@@ -0,0 +1,163 @@
+// Libraries
+import React, {ChangeEvent, FC, useContext, useState} from 'react'
+
+// Contexts
+import {BucketContext} from 'src/shared/contexts/buckets'
+import {DBRPContext} from 'src/shared/contexts/dbrps'
+import {ScriptQueryBuilderContext} from 'src/dataExplorer/context/scriptQueryBuilder'
+import {PersistanceContext} from 'src/dataExplorer/context/persistance'
+
+// Types
+import {RemoteDataState} from 'src/types'
+import {DBRP} from 'src/client'
+import {LanguageType} from 'src/dataExplorer/components/resources'
+
+// Components
+import SelectorTitle from 'src/dataExplorer/components/SelectorTitle'
+import {
+  ComponentSize,
+  Dropdown,
+  DropdownMenuTheme,
+  IconFont,
+  Input,
+  TechnoSpinner,
+} from '@influxdata/clockface'
+
+const DROPDOWN_LABEL: string = 'Database/Retention policy'
+const DBRP_TOOLTIP: string = `InfluxQL requires a database and retention policy \
+(DBRP) combination in order to query data. In InfluxDB Cloud, databases \
+and retention policies have been combined and replaced by InfluxDB buckets. \
+To query InfluxDB Cloud with InfluxQL, the specified DBRP combination \
+must be mapped to a bucket.`
+
+export const DBRPSelector: FC = () => {
+  const {loading, dbrps} = useContext(DBRPContext)
+  const {buckets} = useContext(BucketContext)
+  const {selectedDBRP, selectDBRP} = useContext(ScriptQueryBuilderContext)
+  const [isSearchActive, setIsSearchActive] = useState(false)
+  const [searchTerm, setSearchTerm] = useState('')
+  const {resource} = useContext(PersistanceContext)
+
+  if (resource?.language !== LanguageType.INFLUXQL) {
+    return null
+  }
+
+  const _dbrps: DBRP[] = dbrps.filter(d =>
+    `${d.database}`.toLocaleLowerCase().includes(searchTerm.toLocaleLowerCase())
+  )
+
+  const handleSelectDBRP = (dbrp: DBRP) => {
+    // grab bucket object
+    const bucket = buckets.find(b => b.id === dbrp.bucketID)
+    if (!bucket) {
+      // this should never be happening
+      console.error(
```

Comment:
Yeah, it's a good point! Ideally, this should never happen since each DBRP object has a `bucketID` which is an existing bucket. Since we never know about edge cases, so I updated in this commit https://github.com/influxdata/ui/pull/6648/commits/b1de36b123baf381c3394fb882686716f928a43c to notify users